vm/process/systemVM.go

```diff
@@ -0,0 +1,94 @@
+package process
+
+import (
+	"github.com/ElrondNetwork/elrond-go/vm"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type systemVM struct {
+	systemEI        vm.SystemEI
+	vmType          []byte
+	systemContracts vm.SystemSCContainer
+}
+
+// NewSystemVM instantiates the system VM which is capable of running in protocol smart contracts
+func NewSystemVM(
+	systemEI vm.SystemEI,
+	systemContracts vm.SystemSCContainer,
+	vmType []byte,
+) (*systemVM, error) {
+	if systemEI == nil || systemEI.IsInterfaceNil() {
+		return nil, vm.ErrNilSystemEnvironmentInterface
+	}
+	if systemContracts == nil || systemContracts.IsInterfaceNil() {
+		return nil, vm.ErrNilSystemContractsContainer
+	}
+	if vmType == nil || len(vmType) == 0 {
+		return nil, vm.ErrNilVMType
+	}
+
+	sVm := &systemVM{
+		systemEI:        systemEI,
+		systemContracts: systemContracts,
+		vmType:          make([]byte, len(vmType)),
+	}
+	copy(sVm.vmType, vmType)
+
+	//TODO: run all system smart contracts INIT function at genesis, update roothash and block
+
+	return sVm, nil
+}
+
+// RunSmartContractCreate creates and saves a new smart contract to the trie
+func (s *systemVM) RunSmartContractCreate(input *vmcommon.ContractCreateInput) (*vmcommon.VMOutput, error) {
+	if input == nil {
+		return nil, vm.ErrInputArgsIsNil
+	}
+	if input.CallerAddr == nil {
```

Comment:
Might improve code coverage on this func

Reply:
done